content/articles/creating-movies-and-animations-using-matlab/index.md

```diff
@@ -0,0 +1,243 @@
+
+It did not happen because Matlab noticed that our plot command was inside a `for` loop. Matlab is smart to realize that it will slow down if it draws every single image here in this loop. So it suppresses the plotting until you drop out of the loop and then render the very last seen.
+
+Since that is not the behavior we need here, we will force Matlab to draw the image. We do this by using the `drawnow` function. 
+
+This function forces Matlab to flush the graphics to plot this as it goes.
+
+```matlab
+% Force Matlab to draw the image at this point
+drawnow
+```
+
+Let us now run the code:
+
+![the animated plot](/engineering-education/creating-video-and-movie-animations-using-matlab/animation_three.png)
+
+Now it seems reasonable. Or, we can use the `pause` function. This function takes the pause time as the argument.
+
+```matlab
+pause(0.2)     %It pauses for 0.2 seconds and continues
+```
+What we are doing now is watching flipbooks occur one time on our screen. This isn't exactly what we would like to do here because we want to save the flipbook. 
+
+Let us call the `getframe` function to force the graphics to render and return a bitmap or matrix of the values of the current figure. This function works as the `drawnow` function. 
+
+Comment out the `drawnow` and have the code below:
+
+```matlab
+% Force Matlab to draw the image at this point
+% drawnow
+movieVector = getframe;
+end
+```
+
+When we run this program, every time it hits a point, it will grab the current picture and jam it into the variable `movieVector`. Thus, you will see a vector `movieVector` in the workspace when the program has completed running.
+
+#### Step 5
+The last step is saving the movie. We have a movie vector which is all our flipbook. We need to print it out as an actual `.mp4` file. 
+
+We use a `videoWriter` function that Matlab uses to do a lot of video writing operations. The argument for this function is the name of the movie and, in our case, `curve`.
+
+```matlab
+%% Save the movie
+myWriter = VideoWriter('curve');
+```
+
+Now, let's go ahead and change the parameter, e.g. frame rate
+
+```matlab
+myWriter.FrameRate = 20;    %movie to play at 20frames per second.
+```
+
+We now need to open the video writer, write the movie, and close the file.
+
+```matlab
+open(myWriter);
+writeVideo(myWriter, movieVector);    %It generates a .avi video and saves it in your drive.
+close(myWriter);
+```
+
+To locate your file, you look at the current folder in Matlab and locate it on your device and play it. In case your device cannot open a .avi file, there is an alternative for this. 
+
+The alternative is specifying `.mp4` to the `videoWriter` function, as shown below:
+
+```matlab
+myWriter = VideoWriter('curve', '');
+```
+
+### Conclusion
+Matlab provides a better environment for performing the animations. The in-built function makes this process even quicker. 
```

Comment:
The conclusion needs to be improved to flow better by connecting the points and sentences more. Currently, this is all very seperate.

Such as the example I've provided below which connects your ideas in a fluid way:
Matlab provides a better environment for creating movies and animations because of its in-built functionality that makes the process quicker. Matlab is smart because it performs specific operations automatically such as data generation. Data generation helps visualize scientific ideas especially when it comes to more complex operations.

I hope this tutorial helps you create movies and animations using Matlab. Happy coding.